### ISSUE TITLE
Inheritance should work for all config options, not just `Enabled`.

### ISSUE THREAD

**Issue by pyrogenic** [bbatsov/rubocop#4582]
I would like to have our local dev environment mimic our build environment, in which lint failures fail the build. Unfortunately, it does not seem like I can raise the severity of all `Lint` cops at the same time.

I'm happy to work out the fix myself if such a PR would be accepted. 

## Expected behavior
```yaml
Lint:
  Severity: error
```
Should change the severity for all `Lint` cops to `error`.

## Actual behavior
Setting is silently ignored, while
```yaml
Lint:
  Enabled: false
```
works as expected.

## Steps to reproduce the problem
Add given configuration to `rubocop.yml`.

## RuboCop version
```sh
$ bundle exec rubocop -v
0.47.1
```
